Assets/Mirror/Runtime/NetworkProximityCheck.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+namespace Mirror
+{
+    // unfortunately the name NetworkProximityChecker is already used by our
+    // component that does sphere casts.
+    // maybe rename it later.
+    //
+    // note: we inherit from NetworkBehaviour so we can reuse .netIdentity, etc.
+    // note: unlike UNET, we only allow 1 proximity checker per NetworkIdentity.
+    [DisallowMultipleComponent]
+    public abstract class NetworkProximityCheck : NetworkBehaviour
```

Comment:
It might be better to use an interface since it is just 3 abstract methods

Reply:
DisallowMultipleComponent doesn't work on interfaces